core/src/main/scala/cats/data/Const.scala

```diff
@@ -174,44 +200,49 @@ sealed abstract private[data] class ConstInstances0 extends ConstInstances1 {
         fa.retag[(A, B)].combine(fb.retag[(A, B)])
     }
 
-  implicit def catsDataCommutativeApplicativeForConst[C](implicit
-    C: CommutativeMonoid[C]
-  ): CommutativeApplicative[Const[C, *]] =
-    new ConstApplicative[C] with CommutativeApplicative[Const[C, *]] { val C0: CommutativeMonoid[C] = C }
+  implicit def catsDataContravariantSemigroupalForConst[D: Semigroup]: ContravariantSemigroupal[Const[D, *]] =
+    new ContravariantSemigroupal[Const[D, *]] {
+      override def contramap[A, B](fa: Const[D, A])(f: B => A): Const[D, B] =
+        fa.retag[B]
+      override def product[A, B](fa: Const[D, A], fb: Const[D, B]): Const[D, (A, B)] =
+        fa.retag[(A, B)].combine(fb.retag[(A, B)])
+    }
 }
 
 sealed abstract private[data] class ConstInstances1 extends ConstInstances2 {
 
+  implicit def catsDataCommutativeApplicativeForConst[C](implicit
+    C: CommutativeMonoid[C]
+  ): CommutativeApplicative[Const[C, *]] =
+    new ConstApplicative[C] with CommutativeApplicative[Const[C, *]] {
+      val C0: CommutativeMonoid[C] = C
+    }
+
   implicit def catsDataCommutativeApplyForConst[C](implicit C: CommutativeSemigroup[C]): CommutativeApply[Const[C, *]] =
     new ConstApply[C] with CommutativeApply[Const[C, *]] { val C0: CommutativeSemigroup[C] = C }
 }
 
 sealed abstract private[data] class ConstInstances2 extends ConstInstances3 {
 
-  implicit def catsDataSemigroupForConst[A: Semigroup, B]: Semigroup[Const[A, B]] = _ combine _
-
-  implicit def catsDataPartialOrderForConst[A: PartialOrder, B]: PartialOrder[Const[A, B]] = _ partialCompare _
-
   implicit def catsDataApplicativeForConst[C](implicit C: Monoid[C]): Applicative[Const[C, *]] =
     new ConstApplicative[C] { val C0: Monoid[C] = C }
-}
-
-sealed abstract private[data] class ConstInstances3 extends ConstInstances4 {
-  implicit def catsDataEqForConst[A: Eq, B]: Eq[Const[A, B]] = _ === _
 
   implicit def catsDataApplyForConst[C](implicit C: Semigroup[C]): Apply[Const[C, *]] =
     new ConstApply[C] { val C0: Semigroup[C] = C }
 }
 
-sealed abstract private[data] class ConstInstances4 {
+sealed abstract private[data] class ConstInstances3 extends ConstInstances4 {
 
-  implicit def catsDataFunctorForConst[C]: Functor[Const[C, *]] =
+  @deprecated("Use catsDataTraverseForConst instead", "2.10.0")
```

Comment:
I think we might want to override `map` in `catsDataTraverseForConst`, for Performance:tm:. Or, so long as we are keeping `ConstFunctor`, we can extend from it to inherit the specialized implementation.

Reply:
I made it `final override` because it was also overridden in `ConstApplicative`